code/controllers/stonedmc/subsystem/direction.dm

```diff
@@ -37,54 +39,62 @@ SUBSYSTEM_DEF(direction)
 	if(!resumed)
 		currentrun = deepCopyList(processing_mobs)
 
-	for(var/L in currentrun)
-		var/mob/living/carbon/human/H
-		if(iscarbon(leader_mapping[L]))
-			var/mob/living/carbon/C = leader_mapping[L]
-			while(currentrun[L].len)
-				H = currentrun[L][currentrun[L].len]
-				currentrun[L].len--
-				if(!H)
-					processing_mobs[L].Remove(H)
-					continue
-				H.update_leader_tracking(C)
-				if(MC_TICK_CHECK)
-					return
-		else
-			while(currentrun[L].len)
-				H = currentrun[L][currentrun[L].len]
-				currentrun[L].len--
-				H.clear_leader_tracking()
-				if(MC_TICK_CHECK)
-					return	
-
-/datum/controller/subsystem/direction/proc/start_tracking(squad_id, mob/living/carbon/human/H)
-	if(!H)
+	for(var/squad_id in currentrun)
+		var/mob/living/L
+		var/mob/living/SL = leader_mapping[squad_id]
+		if (QDELETED(SL))
+			clear_run(squad_id) // clear and reset all the squad members
+			continue
+		while(currentrun[squad_id].len)
+			L = currentrun[squad_id][currentrun[squad_id].len]
+			currentrun[squad_id].len--
+			if(QDELETED(L))
+				stop_tracking(squad_id, L)
+				continue
+			L.update_leader_tracking(SL)
+			if(MC_TICK_CHECK)
+				return
+
+/datum/controller/subsystem/direction/proc/clear_run(squad_id)
+	var/mob/living/L
+	while(currentrun[squad_id].len)
+		L = currentrun[squad_id][currentrun[squad_id].len]
+		currentrun[squad_id].len--
+		if(QDELETED(L))

```

Comment:
Make sure on `living/Destroy()` things are removed from the subsystem if they are in it.